plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java

```diff
@@ -424,11 +424,12 @@ private void closeWriter(int writerIndex)
                 writeContext.getPath(),
                 fileFormat,
                 writer.getWrittenBytes(),
-                new MetricsWrapper(writer.getMetrics()),
+                new MetricsWrapper(writer.getFileMetrics().metrics()),
                 PartitionSpecParser.toJson(partitionSpec),
                 writeContext.getPartitionData().map(PartitionData::toJson),
                 DATA,
-                Optional.empty());
+                Optional.empty(),
+                writer.getFileMetrics().splitOffsets());
```

Comment:
Add a test for each file format to make sure that this change is effective.

Reply:
I am verifying that split_offsets are populated in `io.trino.plugin.iceberg.BaseIcebergSystemTables#testFilesTable`